src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11AEADCipher.java

```diff
@@ -401,8 +401,16 @@ private void implInit(int opmode, Key key, byte[] iv, int tagLen,
     }
 
     private void cancelOperation() {
-        // cancel operation by finishing it; avoid killSession as some
-        // hardware vendors may require re-login
+        token.ensureValid();
+        if (!P11Util.trySessionCancel(token, session,
+                (encrypt ? CKF_ENCRYPT : CKF_DECRYPT))) {
+            tryFinishingOff();
+        }
+    }
+
+    // only used by cancelOperation(); cancel by finishing operations
+    // avoid killSession as some hardware vendors may require re-login
```

Comment:
Since this method is only called once, maybe we can inline it back inside `cancelOperation()`?

Reply:
Ok~